src/gui/editorwidgets/core/qgseditorwidgetwrapper.cpp

```diff
@@ -49,6 +49,11 @@ QVariant QgsEditorWidgetWrapper::defaultValue() const
   return mDefaultValue;
 }
 
+void QgsEditorWidgetWrapper::resetValue()
+{
+  setValue( defaultValue() );
```

Comment:
Yes, I think that's the right thing to do (showing a message like "feature does not exist" or so will be considerably harder to get right I guess)